Userland/Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -234,8 +234,14 @@ void ResourceLoader::load(LoadRequest& request, Function<void(ReadonlyBytes, Has
         if (request.page().has_value())
             m_page = request.page().value();
 
-        if (!m_page.has_value())
+        if (!m_page.has_value()) {
+            auto error_message = "No page set for file:// request"sv;
+            log_failure(request, error_message);
+            if (error_callback)
+                error_callback(error_message, {});
             return;
+        }
```

Comment:
Given this is a programming error if it ever occurs and not something the user did wrong (or could fix themselves), let's only `VERIFY()` and crash the process if needed.